src/OpenTelemetry.Instrumentation.AWSLambda/.publicApi/netstandard2.0/PublicAPI.Unshipped.txt

```diff
@@ -0,0 +1,13 @@
+OpenTelemetry.Instrumentation.AWSLambda.AWSLambdaInstrumentationOptions
+OpenTelemetry.Instrumentation.AWSLambda.AWSLambdaInstrumentationOptions.AWSLambdaInstrumentationOptions() -> void
+OpenTelemetry.Instrumentation.AWSLambda.AWSLambdaInstrumentationOptions.DisableAwsXRayContextExtraction.get -> bool
+OpenTelemetry.Instrumentation.AWSLambda.AWSLambdaInstrumentationOptions.DisableAwsXRayContextExtraction.set -> void
+OpenTelemetry.Instrumentation.AWSLambda.AWSLambdaWrapper
+OpenTelemetry.Instrumentation.AWSLambda.TracerProviderBuilderExtensions
+static OpenTelemetry.Instrumentation.AWSLambda.AWSLambdaWrapper.Trace(OpenTelemetry.Trace.TracerProvider tracerProvider, System.Action<Amazon.Lambda.Core.ILambdaContext> lambdaHandler, Amazon.Lambda.Core.ILambdaContext context, System.Diagnostics.ActivityContext parentContext = default(System.Diagnostics.ActivityContext)) -> void
+static OpenTelemetry.Instrumentation.AWSLambda.AWSLambdaWrapper.Trace<TInput, TResult>(OpenTelemetry.Trace.TracerProvider tracerProvider, System.Func<TInput, Amazon.Lambda.Core.ILambdaContext, TResult> lambdaHandler, TInput input, Amazon.Lambda.Core.ILambdaContext context, System.Diagnostics.ActivityContext parentContext = default(System.Diagnostics.ActivityContext)) -> TResult
+static OpenTelemetry.Instrumentation.AWSLambda.AWSLambdaWrapper.Trace<TInput>(OpenTelemetry.Trace.TracerProvider tracerProvider, System.Action<TInput, Amazon.Lambda.Core.ILambdaContext> lambdaHandler, TInput input, Amazon.Lambda.Core.ILambdaContext context, System.Diagnostics.ActivityContext parentContext = default(System.Diagnostics.ActivityContext)) -> void
+static OpenTelemetry.Instrumentation.AWSLambda.AWSLambdaWrapper.TraceAsync(OpenTelemetry.Trace.TracerProvider tracerProvider, System.Func<Amazon.Lambda.Core.ILambdaContext, System.Threading.Tasks.Task> lambdaHandler, Amazon.Lambda.Core.ILambdaContext context, System.Diagnostics.ActivityContext parentContext = default(System.Diagnostics.ActivityContext)) -> System.Threading.Tasks.Task
```

Comment:
It always clarifies things for me to look at this condensed API listing:

Apparently, we have overloads for (very unusual) void-returning handlers with and without `TInput input` parameter. For the (much more common) TResult-returning handlers, we only have the overload with input.

That seems weird to me. Should we remove the non-input overload also for void? Or add the non-input overload for TResult?

Reply:
I would go with "non-input overload for TResult". In case of "removing non-input overload" option we seems force customer to have function handler with redundant unused argument. For example: `TResult Handler(ILambdaContext context, object unusedArgument)`, where `TInput = typeof(object)`.

Reply:
I discovered that the non-input overloads were not originally present but only added in #408 (unreleased).
That makes me think we should remove them after all. My reason being, that having a Lambda without input argument is basically useless, but having the overload may make customers use them nevertheless, since they can just access the parameter passed to the actual handler without passing it through the Trace methods. But that would prevent us from doing special handling for certain inputs

Reply:
Seems handlers without input parameters even not expected:  https://docs.aws.amazon.com/lambda/latest/dg/csharp-handler.html. So, I agree that input-less Trace overloads can be removed.

Reply:
I created #622 to remove the overloads.